Observer/ProductDeleteBefore.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Exception;
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+use Klaviyo\Reclaim\Helper\Webhook;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\ObjectManagerInterface;
+
+
+class ProductDeleteBefore implements ObserverInterface
+{
+    /**
+     * Klaviyo scope setting helper
+     * @var ScopeSetting $klaviyoScopeSetting
+     */
+    protected $_klaviyoScopeSetting;
+
+    /**
+     * @var Webhook $webhookHelper
+     */
+    protected  $_webhookHelper;
+
+    /**
+     * @var ObjectManagerInterface $objectManagerInterface
+     */
+    protected $_objectManager;
+
+    /**
+     * @param Webhook $webhookHelper
+     * @param ObjectManagerInterface $objectManager
+     * @param ScopeSetting $klaviyoScopeSetting
+     */
+    public function __construct(
+        Webhook $webhookHelper,
+        ObjectManagerInterface $objectManager,
+        ScopeSetting $klaviyoScopeSetting
+    ) {
+        $this->_webhookHelper = $webhookHelper;
+        $this->_objectManager = $objectManager;
+        $this->_klaviyoScopeSetting = $klaviyoScopeSetting;
+    }
+
+    /**
+     * customer register event handler
+     *
+     * @param Observer $observer
+     * @return void
+     * @throws Exception
+     */
+    public function execute(Observer $observer)
+    {
+        $product = $observer->getEvent()->getProduct();
+        $websiteIds = $product->getWebsiteIds();
+        $storeIdKlaviyoMap = $this->_klaviyoScopeSetting->getStoreIdKlaviyoAccountSetMap($websiteIds);
```

Comment:
Hi @remstone7,
There is a mistake. You're sending website IDs, while the method receives Store IDs.
Here are more details between them:
https://docs.magento.com/user-guide/configuration/scope.html